src/backend/utils/compatibility_layers.ts

```diff
@@ -0,0 +1,379 @@
+  const wineSet = new Set<WineInstallation>()
+  if (!isMac) {
+    return wineSet
+  }
+  const wineSkinPath = `${userHome}/Applications/Wineskin`
+  if (existsSync(wineSkinPath)) {
+    const apps = readdirSync(wineSkinPath)
+    for (const app of apps) {
+      if (app.includes('.app')) {
+        const wineBin = `${userHome}/Applications/Wineskin/${app}/Contents/SharedSupport/wine/bin/wine64`
+        if (existsSync(wineBin)) {
+          try {
+            const { stdout: out } = await execAsync(`'${wineBin}' --version`)
+            const version = out.split('\n')[0]
+            wineSet.add({
+              ...getWineExecs(wineBin),
+              lib: `${userHome}/Applications/Wineskin/${app}/Contents/SharedSupport/wine/lib`,
+              lib32: `${userHome}/Applications/Wineskin/${app}/Contents/SharedSupport/wine/lib`,
+              name: `Wineskin - ${version}`,
+              type: 'wine',
+              bin: wineBin
+            })
+          } catch (error) {
+            logError(
+              `Error getting wine version for ${wineBin}`,
+              LogPrefix.GlobalConfig
+            )
+          }
+        }
+      }
+    }
+  }
+  return wineSet
+}
+
+/**
+ * Detects CrossOver installs on Mac
+ *
+ * @returns Promise<Set<WineInstallation>>
+ */
+export async function getCrossover(): Promise<Set<WineInstallation>> {
+  const crossover = new Set<WineInstallation>()
+
+  if (!isMac) {
+    return crossover
+  }
+
+  await execAsync(
+    'mdfind kMDItemCFBundleIdentifier = "com.codeweavers.CrossOver"'
+  )
+    .then(async ({ stdout }) => {
+      stdout.split('\n').forEach((crossoverMacPath) => {
+        const infoFilePath = join(crossoverMacPath, 'Contents/Info.plist')
+        if (crossoverMacPath && existsSync(infoFilePath)) {
+          const info = plistParse(
+            readFileSync(infoFilePath, 'utf-8')
+          ) as PlistObject
+          const version = info['CFBundleShortVersionString'] || ''
+          const crossoverWineBin = join(
+            crossoverMacPath,
+            'Contents/SharedSupport/CrossOver/bin/wine'
+          )
+          crossover.add({
+            bin: crossoverWineBin,
+            name: `CrossOver - ${version}`,
+            type: 'crossover',
+            ...getWineExecs(crossoverWineBin)
+          })
+        }
+      })
+    })
+    .catch(() => {
+      logInfo('CrossOver not found', LogPrefix.GlobalConfig)
+    })
+  return crossover
+}
+
+/**
+ * Detects Gaming Porting Toolkit Wine installs on Mac
+ * @returns Promise<Set<WineInstallation>>
+ **/
+export async function getGamingPortingToolkitWine(): Promise<
+  Set<WineInstallation>
+> {
+  const gamingPortingToolkitWine = new Set<WineInstallation>()
+  if (!isMac) {
+    return gamingPortingToolkitWine
+  }
+
+  logInfo('Searching for Gaming Porting Toolkit Wine', LogPrefix.GlobalConfig)
+  await execAsync('mdfind wine64').then(async ({ stdout }) => {
```

Comment:
one concern I have is that at least for me it's hard to reason about async stuff when `async/await` is mixed with `then`

Since we are awaiting for the `execAsync` calls maybe it's better to just use `execSync` instead?